src/App.tsx

```diff
@@ -1,61 +1,128 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
 
 export const App = () => {
+  const [selectedUser, setSelectedUser] = useState('-1');
+  const [userError, setUserError] = useState(false);
+  const [todoTitle, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [todos, setTodos] = useState(todosFromServer);
+  const [titleLanguageError, setTitleLanguageError] = useState(false);
+  const validation = () => {
+    if (selectedUser !== '-1' && todoTitle !== '') {
+      return true;
+    }
+
+    if (selectedUser === '-1') {
+      setUserError(true);
+    }
+
+    if (todoTitle === '') {
+      setTitleError(true);
+    }
+
+    return false;
+  };
+
+  const submitHandler = (event: FormEvent) => {
+    event.preventDefault();
+    if (!validation()) {
+      return;
+    }
+
+    const todoId = todos.reduce((acc, curr) => Math.max(acc, curr.id), -1) + 1;
+
+    const todoToAdd = {
+      id: todoId,
+      title: todoTitle,
+      userId: +selectedUser,
+      completed: false,
+    };
+
+    setSelectedUser('-1');
+    setTitle('');
+
+    setTodos(prevState => [
+      ...prevState,
+      todoToAdd,
+    ]);
+  };
+
   return (
-    <div className="App">
-      <h1>Add todo form</h1>
+    <div className="App panel">
+      <h1
+        className="heading is-size-3"
+      >
+        Add todo form
+      </h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={submitHandler}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
+          <input
+            type="text"
+            className="input"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={todoTitle}
+            onChange={event => {
+              setTitleError(false);
+              if (event.target.value.match(/[А-яа-я]/g)) {
+                setTitleLanguageError(true);
+              } else {
+                setTitleLanguageError(false);
+              }
```

Comment:
_comment_ 
Usually when you want to use if else construction to determine one value, it can be done in a more short form.
If I understand this code right you can simplify it to something like this.
```suggestion
              setTitleLanguageError(event.target.value.match(/[А-яа-я]/g));
```